src/Smalot/PdfParser/PDFObject.php

```diff
@@ -274,14 +275,43 @@ public function getText(Page $page = null)
                         $args = preg_split('/\s/s', $command[self::COMMAND]);
                         $y = array_pop($args);
                         $x = array_pop($args);
+                        // get max char size and use as minimum for horizontal offset
+                        $fontMaxWidth = 0;
+                        if (null !== $current_font) {
+                            $fontDictonary = $current_font->getDetails();
+                            // type 0
+                            if (isset($fontDictonary['DescendantFonts'])) {
+                                $fontDictonary = $fontDictonary['DescendantFonts'][0];
+                            }
+                            // type 1
+                            if (isset($fontDictonary['Widths'])) {
+                                foreach ($fontDictonary['Widths'] as $width) {
+                                    if ((float) $width > $fontMaxWidth) {
+                                        $fontMaxWidth = (float) $width;
+                                    }
+                                }
+
+                                $fontMaxWidth = ($fontMaxWidth / 1000) * $current_font_size;
+                            }
+                            // CIDFontType2
+                            if ('cidfonttype2' === strtolower($fontDictonary['Type'])) {
+                                if (isset($fontDictonary['DW']) && $fontDictonary['DW']) {
+                                    $fontMaxWidth = ((float) $fontDictonary['DW'] / 1000) * $current_font_size;
+                                } else {
+                                    // default
+                                    $fontMaxWidth = $current_font_size;
+                                }
+                            }
+                        }
+
```

Comment:
should be done outside of foreach loop

Reply:
> should be done outside of foreach loop

OK and what do you mean by that? If you see fit, you can change it.

Reply:
yeah just a reminder to myself

Reply:
Ok, then please resolve this conversation if you are done.